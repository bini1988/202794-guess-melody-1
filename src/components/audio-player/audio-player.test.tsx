import React from "react";
import renderer from "react-test-renderer";
import serializer from "jest-emotion";
import AudioPlayer from "./audio-player";

expect.addSnapshotSerializer(serializer);

it(`AudioPlayer correctly renders default markup`, () => {
  const tree = renderer
    .create(
        <AudioPlayer
          src=""/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
