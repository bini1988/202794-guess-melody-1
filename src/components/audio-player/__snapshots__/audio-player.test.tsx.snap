// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer correctly renders default markup 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  padding: 0;
  width: 35px;
  height: 50px;
  border: none;
  background-color: transparent;
  -webkit-transition: -webkit-transform ease-in-out 300ms;
  -webkit-transition: transform ease-in-out 300ms;
  transition: transform ease-in-out 300ms;
  outline: none;
  cursor: pointer;
}

.emotion-1:not(:disabled):hover,
.emotion-1:not(:disabled):focus {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.emotion-1:not(:disabled):active {
  -webkit-transform: scale(1.04);
  -ms-transform: scale(1.04);
  transform: scale(1.04);
}

.emotion-1 svg {
  fill: #ff9749;
  -webkit-filter: drop-shadow(3px 3px 3px rgba(0,0,0,0.2));
  filter: drop-shadow(3px 3px 3px rgba(0,0,0,0.2));
}

.emotion-0 {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  border: 0;
  padding: 0;
  white-space: nowrap;
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  overflow: hidden;
}

.emotion-3 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  margin: 0 10px;
  height: 55px;
  width: 370px;
}

.emotion-3 > svg {
  -webkit-filter: drop-shadow(5px 5px 5px rgba(0,0,0,0.2));
  filter: drop-shadow(5px 5px 5px rgba(0,0,0,0.2));
}

<div
  className="emotion-5 emotion-6"
>
  <button
    className="emotion-1 emotion-2"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      height="50"
      width="35"
    >
      <use
        xlinkHref="#play"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
    <span
      className="emotion-0"
    >
      Проиграть трек
    </span>
  </button>
  <div
    className="emotion-3 emotion-4"
  >
    <svg
      height="55"
      width="370"
    >
      <use
        xlinkHref="#equalizer"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
      <rect
        clipPath="url(#equalizer-lines)"
        fill="#ff9749"
        height="100%"
        width="0%"
      />
    </svg>
    <audio
      onCanPlayThrough={[Function]}
      onEnded={[Function]}
      onError={[Function]}
      onLoadStart={[Function]}
      onTimeUpdate={[Function]}
      src=""
    />
  </div>
</div>
`;
